torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -529,6 +535,10 @@ class ShapePropagator {
         return;
       }
       case aten::_unwrap_optional: {
+	if (node->input()->type()->isSubtypeOf(TensorType::get())) {
```

Comment:
Since _unwrap_optional will throw an exception if it's None, we can do the same thing as with prim::unchecked_unwrap and set the output to be the unwrapped type of the input if it's an optional input, and the input if it's not an optional

Reply:
Same as above. Actually, the None handling is already in that case.